presto-tests/src/test/java/com/facebook/presto/resourcemanager/TestDistributedQueryResource.java

```diff
@@ -156,4 +156,14 @@ private void assertStateCounts(List<BasicQueryInfo> infos, int expectedFinished,
         assertEquals(running, expectedRunning);
         assertEquals(queued, expectedQueued);
     }
+
+    @Test
+    public void testGetAllQueryInfoForLimits() {
+        runToCompletion(client, coordinator1, "SELECT 1");
+        runToFirstResult(client, coordinator1, "SELECT * from tpch.sf100.orders");
+        runToFirstResult(client, coordinator1, "SELECT * from tpch.sf101.orders");
+        runToFirstResult(client, coordinator1, "SELECT * from tpch.sf102.orders");
+        List<BasicQueryInfo> infos = getQueryInfos(client, coordinator1, "/v1/query?limit=1");
```

Comment:
let's use limit > 1.

Reply:
Also let's add a test related to sorting where a queued/running queries and see if running queries return first or not.

Reply:
Also before checking for limit, make sure all queries are being returned
i.e. /v1/query to return 4 queries.